python3Packages.geojson-pydantic: init at 2.1.1

deej-io opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

This PR adds myself as a maintainer and packages the geojson-pydantic python library.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

deej-io committed:
maintainers: add deej-io

deej-io committed:
python312Packages.geojson-pydantic: init at 2.0.0

deej-io committed:
python3Packages.geojson-pydantic: init at 2.1.1

thunze commented on pkgs/development/python-modules/geojson-pydantic/default.nix:
```diff
@@ -0,0 +1,41 @@
+    hash = "sha256-SOo4Hs1WMUerMLMgjzUVuDkUXCuiStJ4P9iMSpUF8Uw=";
+  };
+
+  nativeBuildInputs = [ flit-core ];
```
```suggestion
  build-system = [ flit-core ];
```

thunze commented on pkgs/development/python-modules/geojson-pydantic/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+  meta = {
+    changelog = "https://github.com/developmentseed/geojson-pydantic/blob/${version}/CHANGELOG.md";
+    description = "geojson_pydantic provides a suite of Pydantic models matching the GeoJSON specification rfc7946. Those models can be used for creating or validating geojson data";
```
```suggestion
    description = "Suite of Pydantic models matching the GeoJSON specification RFC 7946";
```

https://nixos.org/manual/nixpkgs/stable/#var-meta-description

If you want to add more information, feel free to use [`longDescription`](https://nixos.org/manual/nixpkgs/stable/#var-meta-longDescription).

thunze commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 415886`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>nixpkgs-manual</li>
    <li>python312Packages.geojson-pydantic</li>
    <li>python312Packages.geojson-pydantic.dist</li>
    <li>python313Packages.geojson-pydantic</li>
    <li>python313Packages.geojson-pydantic.dist</li>
  </ul>
</details>

deej-io commented:
Is there anything else that needs to be for this to be merged?

autra reviewed (changes requested):
Thanks for this addition!

2 little changes and we're good to go imo.

thunze commented on pkgs/development/python-modules/geojson-pydantic/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    shapely
```
After taking a closer look, the [Python docs](https://nixos.org/manual/nixpkgs/unstable/#buildpythonpackage-function) actually recommend using `nativeCheckInputs`:

> Test dependencies, like e.g. the test runner, should be added to `nativeCheckInputs`.

I asked on Matrix (Python room) about why we use `nativeCheckInputs` but unfortunately got no conclusive answer.

> and I think that people used to put test dependencies in `nativeCheckInputs` because there was a bug in checkInputs. This bug has since been fixed but inertia remains.

Are you referring to https://github.com/NixOS/nixpkgs/issues/161570? Because back when that issue was created, `nativeCheckInputs` wasn't a thing. In fact, https://github.com/NixOS/nixpkgs/pull/206742 (https://github.com/NixOS/nixpkgs/commit/33afbf39f6f2a6b37e99f070ba7d17a28c416d02) switched all usages of `checkInputs` to `nativeCheckInputs` because `checkInputs` used to behave like `nativeCheckInputs`.

In any case, I don't think this is worth blocking over in a review as it doesn't seem to provide any measurable benefit (I'm happy to be proven wrong on this!) and contradicts the Python docs.

autra replied:
I'm sorry but I still have a different interpretation. The sentence you quote says "like eg. the test runner" (like pytest, which is a cli tool and must be added to the PATH) and other part of the doc makes it clear that `native*` is for `PATH`, so executable, not dependencies. Ex for the stdenv doc

> nativeCheckInputs for test tools needed on $PATH (such as ctest) and [setup hooks](https://nixos.org/manual/nixpkgs/unstable/#ssec-setup-hooks) (for example [pytestCheckHook](https://nixos.org/manual/nixpkgs/unstable/#python))
> 
> checkInputs for libraries linked into test executables (for example the qcheck OCaml package)

Imo, linkage is the equivalent of import for the python world.

I don't understand your point about #161570 and #206742 because the description of the latter seems to be consistent with my interpretation.

> 
> tl;dr: checkInputs used to be added to nativeBuildInputs, which means they are added to PATH but one cannot link to them when strictDeps is set. As linking to check-only libs is possible (quite common for ocaml for example) we need to distinguish
> 
>     checkInputs, added to buildInputs when doCheck is set. One can link against them.
>     nativeCheckInputs, added to nativeBuildInputs when doCheck is set. They are added to PATH.
>     (same for installCheck)



> In any case, I don't think this is worth blocking over in a review as it doesn't seem to provide any measurable benefit

Benefits: consistency, semantics and so less likely to become a problem later on. Both works today, but may not in the future.

I agree it's not worth to block on this, so if any other maintainer approves, it's fine by me :-)

thunze replied:
I encourage you to open an issue to discuss this further. Maybe some contributors with relevant background knowledge can clarify this :)

autra commented:
> I agree it's not worth to block on this, so if any other maintainer approves, it's fine by me :-)

Anyhow, you're right, let's not argue too much about it. Choose what you prefer on this native vs non native thing, let's add the geospatial team in the meta as well, and I'll approve.

autra commented on pkgs/development/python-modules/geojson-pydantic/default.nix:
```diff
@@ -0,0 +1,41 @@
+    description = "Suite of Pydantic models matching the GeoJSON specification RFC 7946";
+    homepage = "https://github.com/developmentseed/geojson-pydantic";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ deej-io ];
```
@deej-io do you plan to finish this?

deej-io commented:
Apologies for the delay.

I moved shapely to checkInputs and added the geospatial team as requested. I’ve rebased this onto current master and updated geojson-pydantic to 2.1.1.
